controller/users.go

```diff
@@ -215,9 +215,43 @@ func (c *UsersController) Update(ctx *app.UpdateUsersContext) error {
 func (c *UsersController) List(ctx *app.ListUsersContext) error {
 	return application.Transactional(c.db, func(appl application.Application) error {
 		var err error
-		var users []*account.User
 		var result *app.UserArray
-		users, err = appl.Users().List(ctx.Context)
+
+		if ctx.Username != nil {
+			// From a data model perspective, we are querying by identity ( and not user )
+			identities, err := appl.Identities().Query(account.IdentityFilterByUsername(*ctx.Username))
+
+			if err != nil {
+				return jsonapi.JSONErrorResponse(ctx, errors.Wrap(err, "error listing users"))
```

Comment:
maybe the error message could be more specific and detail more about the failed operation